### PR TITLE
Include metadata keys by default for batch processor

### PR DESCRIPTION
**Pull request by asreehari-splunk**
**Description:** Update the default setting of the batch processor to retrieve token from the context using the key

**Link to Splunk idea:** <Link to Splunk idea, see https://ideas.splunk.com>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl [hunk 1]
```diff
@@ -82,6 +82,8 @@ processors:
   {{- end }}
 
   batch:
+    metadata_keys:
+      - X-SF-Token
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl:86]: This is the only configuration where we don't need it. The cluster receiver doesn't get any push-based receivers.

File: .chloggen/add_default_metadata_key_batch_processor.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+# The name of the component, or a single word describing the area of concern, (e.g. agent, clusterReceiver, gateway, operator, chart, other)
+component: template
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: add default metadata key for token to batch processor
+# One or more tracking issues related to the change
+issues: []
```
Comment (dmitryax) [inline .chloggen/add_default_metadata_key_batch_processor.yaml:8]:
```suggestion
issues: [1467]
```

**Review by atoulme — APPROVED** [review]

File: .chloggen/add_default_metadata_key_batch_processor.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+# The name of the component, or a single word describing the area of concern, (e.g. agent, clusterReceiver, gateway, operator, chart, other)
+component: template
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: add default metadata key for token to batch processor
```
Comment (jvoravong) [inline .chloggen/add_default_metadata_key_batch_processor.yaml:6]:
```suggestion
note: Add default metadata key for token to batch processor
```

**Review by jvoravong — APPROVED** [review]
Merge in the two suggestions for the .chloggen file and this PR LGTM.

**Pull request merged**